backend/candidate_comparison_tool/service.py

```
@@ -0,0 +1,31 @@
+from .prompts import build_prompt
+
+genai.configure(api_key=os.getenv("GEMINI_API_KEY"))
+
+
+def extract_json(text):
+    match = re.search(r"\{.*\}", text, re.DOTALL)
+    if match:
+        try:
+            return json.loads(match.group(0))
+        except json.JSONDecodeError as e:
+            raise ValueError(f"Failed to parse JSON: {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)